src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -553,12 +553,18 @@ private[internal] trait GlbLubs {
     val ntps: List[PolyType]   = ptHead :: ptRest.map(normalizeIter)
 
     val tparams1: List[Symbol] = {
-      def unifyBounds(ntp: PolyType): List[Type] = {
+      val substSymMaps: Array[SubstSymMap] = new Array[SubstSymMap](ptRest.length)
+      foreachWithIndex(ntps.tail) { (ntp, ix) =>
         val tparams1 = ntp.typeParams
-        tparams1 map (tparam => tparam.info.substSym(tparams1, tparamsHead))
+        if (! ((tparams1 eq tparamsHead) || tparams1.isEmpty))
```

Comment:
This optimisation is from the code of `substSym`, but perhaps the case they describe would not apply here, so can be removed? It's not a major problem, though.